readme.md

```diff
@@ -35,14 +35,30 @@ require('./foo')();
 
 ## API
 
-### clearModule(moduleId)
+### clearModule(moduleId, options)
 
 #### moduleId
 
 Type: `string`
 
 What you would use with `require()`.
 
+#### options
+
+Type: `object`
+
+##### options.filter
+
+Optional callback to filter with, must return false to not clear. Passes in the full path to the module. Will not process child modules of modules that are filtered out.
+
+Use case: This allows you to prevent specific submodules from being unloaded, you may need to do this if modules are shared between always running code, and modules that are loaded/unloaded.
+
+`clearModule('./my-module', { filter: name => !name.match(/aws-sdk/) })` would prevent the aws-sdk from being cleared
+
+`clearModule('./my-module', { filter: name => !name.match(/node_modules/) })` would prevent all node_modules from being cleared
+
+Note: not the same as the match function, as the match function still clears all children of the match.
```

Comment:
The readme and index.d.ts should be in sync docs-wise.

Reply:
I think the whole text could be a little bit better written.

Reply:
Synced the documentation with index.d.ts, I removed the doc above.  I'm not sure what you would like here, you asked for a "use case" in the documentation, but there is no example of this in the current documentation.  I'm not sure what you want here or how you want it worded.